src/polyhedralGravity/model/SplitNode.cpp

```diff
@@ -0,0 +1,108 @@
+#include "SplitNode.h"
+
+#include "KDTree.h"
+#include "KdDefinitions.h"
+#include "TreeNodeFactory.h"
+
+#include <algorithm>
+
+namespace polyhedralGravity {
+
+    SplitNode::SplitNode(const SplitParam &splitParam, Plane &plane, TriangleIndexLists<2> &triangleIndexLists)
+        : TreeNode(splitParam), _plane{plane}, _triangleIndexLists{std::move(triangleIndexLists)} {
+        //convert the bound faces to its vertices
+        std::vector<Array3> boundVertices{};
+        boundVertices.reserve(splitParam.indexBoundFaces.size() * 3);
+        std::for_each(splitParam.indexBoundFaces.cbegin(), splitParam.indexBoundFaces.cend(), [&boundVertices, &splitParam](const size_t faceIndex) {
+            for (const auto &vertexIndex: splitParam.faces[faceIndex]) {
+                boundVertices.push_back(splitParam.vertices[vertexIndex]);
+            }
+        });
+        _boundingBox = KDTree::getBoundingBox(boundVertices);
+    }
+
+    std::shared_ptr<TreeNode> SplitNode::getLesserNode() {
+        if (!this->_lesser) {                                                                                                         //node is not yet built
+            SplitParam childParam{*this->splitParam};                                                                                 //copy parent param and modify to fit new node
+            childParam.boundingBox = (KDTree::splitBox(this->_boundingBox, this->_plane)).first;                                      //get the lesser box after splitting;
+            childParam.indexBoundFaces = *std::move(_triangleIndexLists[0]);                                                          //get the triangles of the lesser box
+            childParam.splitDirection = static_cast<Direction>((static_cast<int>(this->splitParam->splitDirection) + 1) % DIMENSIONS);//set the next splitting dimension: begin: (X -> Y -> Z); goto begin;
+            _lesser = TreeNodeFactory::treeNodeFactory(childParam);
+            maybeFreeParam();//if both nodes are built, split parameters are no longer needed
+        }
+        return _lesser;
+    }
+
+    std::shared_ptr<TreeNode> SplitNode::getGreaterNode() {
```

Comment:
Code Duplication to `getLesserNode`, consider parameterizing or explicit templating (this works in the source files as well)